healthScore/views.py

```diff
@@ -395,53 +397,79 @@ def login_view(request):
     return render(request, "login.html")
 
 
+@login_required
 def view_health_history_requests(request):
     zipped_details = get_health_history_details(request=request)
 
     return render(request, "view_requests.html", {"zipped_details": zipped_details})
 
 
 @login_required
-@csrf_exempt
-def get_hospitals(request):
-    hospitalList = list(Hospital.objects.all().values())
-    data = {
-        "hospitals": hospitalList,
-        "appointmentType": APPOINTMENT_TYPE,
-        "appointmentProps": json.dumps(APPOINTMENT_PROPS),
-    }
-    return render(request, "submit_health_record.html", {"data": data})
+def record_sent_view(request):
+    return render(request, "record_submit_complete.html")
 
 
-@login_required
-@csrf_exempt
 def get_doctors(request, hos_id):
     doctorList = list(
         HospitalStaff.objects.filter(admin=False, hospitalID_id=hos_id).values()
     )
     return JsonResponse({"doctors": doctorList})
 
 
-# def create_record(request):
-#     print(request, "request")
-#     updatedData = json.loads(request.body)
-#     current_user = ""
-#     current_user = request.user
-
-#     current_user_id = current_user.id
-
-#     new_record = Appointment.objects.create(
-#         name=updatedData['appointmentType'],
-#         properties=updatedData["appointmentProperties"]
-#     )
+@login_required
+def add_health_record_view(request):
+    hospitalList = list(Hospital.objects.all().values())
```

Comment:
We can maybe have these lines after the if condition. This way we would be able to avoid the unnecessary DB calls for the other method types

Reply:
This part is from your code. My interpretation is that you want to make sure no matter what method, props should be displayed. That's why it is out of the POST method. Correct me if I am wrong.

Reply:
Oh right right. My bad. I didn't see the return statements here. You are right!